get_ecg.py

```diff
@@ -0,0 +1,173 @@
+            dump = []
+            new_time_array = divided_time_array[i]
+            new_voltage_array = divided_voltage_array[i]
+            max_peaks = []
+
+            mx, mn = -np.Inf, np.Inf
+
+            for index, (x, y) in enumerate(zip(new_time_array,
+                                               new_voltage_array)):
+                if y > mx:
+                    mx = y
+                    mxpos = x
+                if y < mn:
+                    mn = y
+
+                # Looking for max
+                if y < mx and mx != np.Inf:
+                    # Maxima peak candidate found
+                    if new_voltage_array[index:index + min_dist].max() < mx:
+                        max_peaks.append([mxpos, mx])
+                        dump.append(True)
+                        # set algorithm to only find minima now
+                        mn = np.Inf
+                        mx = np.Inf
+                        if index + min_dist >= length:
+                            # end is within lookahead no more peaks
+                            break
+                        continue
+
+                if y > mn and mn != -np.Inf:
+                    if new_voltage_array[index:index + min_dist].min() > mn:
+                        dump.append(False)
+                        mn = -np.Inf
+                        mx = -np.Inf
+                        if index + min_dist >= length:
+                            break
+
+            self.total_peaks.append(max_peaks)
+            # Remove the false hit on the first value of the y_axis
+            try:
+                if dump[0]:
+                    max_peaks.pop(0)
+                del dump
+            except IndexError:
+                    # no peaks were found
+                    print("No peaks were found")
+
+        return self.total_peaks
+
+    def get_inst_hr(self):
+        """
+           Function that calculates inst heart rate every update_time seconds
+           by counting number of peaks within update_time
+
+           :return self.raw_bunches: divided the total_peaks into custom peaks
+           :rtype: nd array
+
+           """
+
+        num_of_peaks = []
+        if isinstance(self.update_time, float):
+            self.update_time = int(self.update_time)
+        for i in range(len(self.total_peaks)):
+            num_of_peaks.append(len(self.total_peaks[i]))
+        new_array = np.array(num_of_peaks)
+        inst_heart_rate = new_array / self.update_time
+        self.raw_bunches = inst_heart_rate * 60
+        return self.raw_bunches
+
+    def get_avghr(self):
+        """ returns avghr
+
+            Function gets avghr over user-specified minutes window.
+            Also gives clinical indication of tachy/bradycardia
+
+            :return avg_hr: Avg hr
+            :return status: Brady or tachycardia
+        """
+        mins = float(input('Please specify a time (in min) for averaging.'))
```

Comment:
What happens if you don't want to run your code interactively?

Reply:
Ideally this function should be reusable by another developer who may want the avghr without needing interactive capabilities

Reply:
Yeah, I mistook user-specified minutes for interaction rather than just setting a default.  Thanks!